+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+            %              casadi symbolics vs just a vectorized assignment which does one. As such there needs to be backend
+            %              work done for vdx to cache vertcats of SX somehow. Current theory is one can simply keep a queue of
+            %              symbolics to be added in a cell array until a read is done, at which point we call a single vertcat
+            %              on the whole queue which is _significantly_ faster.
+            % 2d vars: Variables that are defined for each finite element.
+            for ii=1:opts.N_stages
+                % other derived values
+                h0 = opts.h_k(ii);
+                if obj.opts.use_fesd
+                    ubh = (1 + opts.gamma_h) * h0;
+                    lbh = (1 - opts.gamma_h) * h0;
+                    if opts.time_rescaling && ~opts.use_speed_of_time_variables
+                        % if only time_rescaling is true, speed of time and step size all lumped together, e.g., \hat{h}_{k,i} = s_n * h_{k,i}, hence the bounds need to be extended.
+                        ubh = (1+opts.gamma_h)*h0*opts.s_sot_max;
+                        lbh = (1-opts.gamma_h)*h0/opts.s_sot_min;
+                    end
+                    obj.w.h(ii,1:opts.N_finite_elements(ii)) = {{'h', 1}, lbh, ubh, h0};
+                end
+                if obj.opts.step_equilibration == StepEquilibrationMode.linear_complementarity
+                    obj.w.B_max(ii,2:opts.N_finite_elements(ii)) = {{'B_max', dims.n_lambda},-inf,inf};
+                    obj.w.pi_c(ii,2:opts.N_finite_elements(ii)) = {{'pi_c', dims.n_c},-inf,inf};
+                    obj.w.pi_lambda(ii,2:opts.N_finite_elements(ii)) = {{'pi_lambda', dims.n_lambda},-inf,inf};
+                    obj.w.lambda_c(ii,2:opts.N_finite_elements(ii)) = {{'lambda_c', dims.n_c},0,inf};
+                    obj.w.lambda_lambda(ii,2:opts.N_finite_elements(ii)) = {{'lambda_lambda', dims.n_lambda},0,inf};
+                    obj.w.eta(ii,2:opts.N_finite_elements(ii)) = {{'eta', dims.n_lambda},0,inf};
+                    obj.w.nu(ii,2:opts.N_finite_elements(ii)) = {{'nu', 1},0,inf};
+                end
+                if ~opts.right_boundary_point_explicit
+                    obj.w.not_c_primals(ii,1:opts.N_finite_elements(ii)) = {{'not_c_primal', 2}, 0 inf, 0.5};
+                    obj.w.mu_not_c(ii,1:opts.N_finite_elements(ii)) = {{'mu_lambda', 3}, 0 inf, 0.5};
+                end
+            end
+
+            % For c_n ~= 1 case
+            rbp = ~opts.right_boundary_point_explicit;
+            
+            % 3d vars: Variables defined on each rk stage
+            %          some of which are also defined at the initial point:
+            obj.w.x(0,0,opts.n_s) = {{['x_0'], dims.n_x}, model.x0, model.x0, model.x0};
+            obj.w.z(0,0,opts.n_s) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
```

Comment:
this are user algebraics? lifting algebraic, sdf algebarics? 
i always have to stop and think whenever i see z. adding comments in few occasions (also in other .m files than this would help our present and future self)